### PR TITLE
Feature - TanHuynh - LoginScreen

### PR DESCRIPTION
**Pull request by at-tanhuynh**

### CODE REVIEW

**Comment by at-lamle** [comment]
:+1: There are no linter violations.

File: MyApp/Model/Session.swift [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+//  Copyright © 2019 Asian Tech Co., Ltd. All rights reserved.
+//
+
+import Foundation
+
+final class Session {
+
+    private let userDefault = UserDefaults.standard
+    static let shared = Session()
+
```
Comment (at-hainguyen3) [inline MyApp/Model/Session.swift]: Thêm `private init() {}`

File: MyApp/View/AppDelegate.swift [hunk 2]
```diff
@@ -26,13 +32,28 @@ final class AppDelegate: UIResponder, UIApplicationDelegate {
...
         configNetwork()
         return true
     }
 
+    func setRootViewController(root: Root) {
+        switch root {
+        case .home:
+            let tabbarVC = TabBarViewController()
+            window?.rootViewController = tabbarVC
+
```
Comment (at-hainguyen3) [inline MyApp/View/AppDelegate.swift]: Remove line space

File: MyApp/View/Controller/Login/WebView/WebViewController.swift [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+    func webView(_ webView: WKWebView, decidePolicyFor navigationAction: WKNavigationAction, decisionHandler: @escaping (WKNavigationActionPolicy) -> Void) {
+        if let url = navigationAction.request.url {
+            if let code = url.getQueryStringParameter(key: "code"), url.absoluteString.hasPrefix("https://www.google.com") {
+                let url = viewModel.urlForLoadWebView(type: .accessToken(code: code))
+                do {
+                    let request = try URLRequest(url: url.asURL())
+                    webView.load(request)
+                } catch {
+                    print(error)
```
Comment (at-hainguyen3) [inline MyApp/View/Controller/Login/WebView/WebViewController.swift]: show alert luôn

File: MyApp/View/Controller/Login/WebView/WebViewModel.swift [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+//  Copyright © 2019 Asian Tech Co., Ltd. All rights reserved.
+//
+
+import Foundation
+import MVVM
+
+final class WebViewModel: ViewModel {
+
+    enum Request {
+        case code
```
Comment (at-hainguyen3) [inline MyApp/View/Controller/Login/WebView/WebViewModel.swift]: để `authenticate` dễ hiểu hơn

**Pull request merged**